### ISSUE TITLE
node attributes are not working with chef-client --local-mode

### ISSUE THREAD

**Issue by smford22** [chef/chef#2959]
### Version:
```
[~/chef-repo/cookbooks/with_data_center_test] (server_cluster)
scottford$ chef --version
Chef Development Kit Version: 0.4.0
```

### Environment: [Details about the environment such as the Operating System, cookbook details, etc...]
```
    Hardware Overview:

      Model Name: MacBook Pro
      Model Identifier: MacBookPro11,1
      Processor Name: Intel Core i7
      Processor Speed: 2.8 GHz
      Number of Processors: 1
      Total Number of Cores: 2
      L2 Cache (per Core): 256 KB
      L3 Cache: 4 MB
      Memory: 16 GB
      Boot ROM Version: MBP111.0138.B03
      SMC Version (system): 2.16f58
```

```
Software:

    System Software Overview:

      System Version: OS X 10.9.5 (13F34)
      Kernel Version: Darwin 13.4.0
      Time since boot: 2:22
```

### Scenario:
I am trying to use node attributes with chef-client --local-mode but that produces an error `ERROR: undefined method `[]' for nil:NilClass`

### Steps to Reproduce:

Create a test cookbook 
`chef generate cookbook attribute test`

Add a basic node attribute to `attributes/default.rb`
```
scottford$ cat attributes/default.rb
default['attribute_test']['name'] = 'scott ford'
```

Add a basic `puts` statement in the default recipe to print the node attribute:
```
scottford$ cat recipes/default.rb
#
# Cookbook Name:: attribute_test
# Recipe:: default
#
# Copyright (C) 2014
#
#
#

puts "DEFAULT NAME: #{node['attribute_test']['name']}"

```

Run chef-client in local mode
```
scottford$ chef-client -z recipes/default.rb
Starting Chef Client, version 12.0.3
resolving cookbooks for run list: []
[2015-02-19T15:46:22-08:00] WARN: Cookbook 'logrotate' is empty or entirely chefignored at /Users/scottford/chef-repo/cookbooks/logrotate
Synchronizing Cookbooks:
Compiling Cookbooks...

Running handlers:
[2015-02-19T15:46:22-08:00] ERROR: Running exception handlers
Running handlers complete
[2015-02-19T15:46:22-08:00] ERROR: Exception handlers complete
[2015-02-19T15:46:22-08:00] FATAL: Stacktrace dumped to /Users/scottford/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
Chef Client failed. 0 resources updated in 2.6736 seconds
[2015-02-19T15:46:22-08:00] ERROR: undefined method `[]' for nil:NilClass
[2015-02-19T15:46:22-08:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

Here is the stacktrace out:
```
scottford$ cat /Users/scottford/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
Generated at 2015-02-19 15:46:22 -0800
NoMethodError: undefined method `[]' for nil:NilClass
/Users/scottford/chef-repo/cookbooks/attribute_test/recipes/default.rb:10:in `from_file'
/opt/chefdk/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chefdk/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chefdk/embedded/apps/chef/lib/chef/run_context.rb:180:in `load_recipe_file'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:76:in `block in setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:75:in `each'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:75:in `setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/client.rb:235:in `setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/client.rb:397:in `run'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:261:in `block in fork_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:249:in `fork'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:249:in `fork_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:215:in `block in run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:201:in `run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:355:in `block in interval_run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:345:in `loop'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:345:in `interval_run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:335:in `run_application'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:58:in `run'
/opt/chefdk/embedded/apps/chef/bin/chef-client:26:in `<top (required)>'
/opt/chefdk/bin/chef-client:40:in `load'
/opt/chefdk/bin/chef-client:40:in `<main>'
```

### Expected Result:
I am expecting chef-client to print out `DEFAULT NAME: scott ford`

### Actual Result:
I get the Undefined Method error on line 10 of the default.rb recipe which is the puts statement. 
```
scottford$ chef-client -z recipes/default.rb
Starting Chef Client, version 12.0.3
resolving cookbooks for run list: []
[2015-02-19T15:46:22-08:00] WARN: Cookbook 'logrotate' is empty or entirely chefignored at /Users/scottford/chef-repo/cookbooks/logrotate
Synchronizing Cookbooks:
Compiling Cookbooks...

Running handlers:
[2015-02-19T15:46:22-08:00] ERROR: Running exception handlers
Running handlers complete
[2015-02-19T15:46:22-08:00] ERROR: Exception handlers complete
[2015-02-19T15:46:22-08:00] FATAL: Stacktrace dumped to /Users/scottford/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
Chef Client failed. 0 resources updated in 2.6736 seconds
[2015-02-19T15:46:22-08:00] ERROR: undefined method `[]' for nil:NilClass
[2015-02-19T15:46:22-08:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

Here is the stacktrace out:
```
scottford$ cat /Users/scottford/chef-repo/.chef/local-mode-cache/cache/chef-stacktrace.out
Generated at 2015-02-19 15:46:22 -0800
NoMethodError: undefined method `[]' for nil:NilClass
/Users/scottford/chef-repo/cookbooks/attribute_test/recipes/default.rb:10:in `from_file'
/opt/chefdk/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `instance_eval'
/opt/chefdk/embedded/apps/chef/lib/chef/mixin/from_file.rb:30:in `from_file'
/opt/chefdk/embedded/apps/chef/lib/chef/run_context.rb:180:in `load_recipe_file'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:76:in `block in setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:75:in `each'
/opt/chefdk/embedded/apps/chef/lib/chef/policy_builder/expand_node_object.rb:75:in `setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/client.rb:235:in `setup_run_context'
/opt/chefdk/embedded/apps/chef/lib/chef/client.rb:397:in `run'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:261:in `block in fork_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:249:in `fork'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:249:in `fork_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:215:in `block in run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:201:in `run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:355:in `block in interval_run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:345:in `loop'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:345:in `interval_run_chef_client'
/opt/chefdk/embedded/apps/chef/lib/chef/application/client.rb:335:in `run_application'
/opt/chefdk/embedded/apps/chef/lib/chef/application.rb:58:in `run'
/opt/chefdk/embedded/apps/chef/bin/chef-client:26:in `<top (required)>'
/opt/chefdk/bin/chef-client:40:in `load'
/opt/chefdk/bin/chef-client:40:in `<main>'
```

**Comment by lamont-granquist** [comment]
This may have been due to breakage in attributes early in 12.0.x which were largely all fixed by 12.2.0

**Comment by adamedx** [comment]
@smford22, I'm not able to repro with chefdk 0.6.0. I recreated your cookbook, and the only thing I did differently was to make sure I placed the cookbook in a cookbooks subdirectory and ran chef-client from the directory that contains the cookbooks subdirectory. I also used `--disable-config` to make sure I didn't pick up any strange config settings from a .chef somewhere:

```
seaedwards01:test adamedwards$ chef exec chef-client -z -o 'attribute' --disable-config
[2015-06-17T10:50:40-07:00] WARN: No config file found or specified on command line, using command line options.
Starting Chef Client, version 12.3.0
[2015-06-17T10:50:42-07:00] WARN: Run List override has been provided.
[2015-06-17T10:50:42-07:00] WARN: Original Run List: []
[2015-06-17T10:50:42-07:00] WARN: Overridden Run List: [recipe[attribute]]
resolving cookbooks for run list: ["attribute"]
Synchronizing Cookbooks:
  - attribute
Compiling Cookbooks...
DEFAULT NAME: scott ford
Converging 0 resources
[2015-06-17T10:50:42-07:00] WARN: Skipping final node save because override_runlist was given

Running handlers:
Running handlers complete
Chef Client finished, 0/0 resources updated in 2.670805 seconds
```
